### PR TITLE
IGNITE-18241 Introduce support code for LogicalTopologyService

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-18241

Later, it is planned to make `LogicalTopologyServiceImpl` also implement new (not yet existing) interface `LogicalTopologyService` that will allow to subscribe to logical topology-related events.

### CODE REVIEW

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/topology/LogicalTopologyImpl.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import java.util.Collection;
+import java.util.Set;
+import org.apache.ignite.internal.cluster.management.raft.ClusterStateStorage;
+import org.apache.ignite.network.ClusterNode;
+
+/**
+ * Implementation of {@link LogicalTopology}.
+ */
+public class LogicalTopologyImpl implements LogicalTopology {
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/topology/LogicalTopologyImpl.java:32]: Before the previous PR, we didn't; before this PR, we did. This caching looks like an optimization and it's not clear whether it's needed at all (all operations with logical topology seem to be pretty rare).

File: modules/rest/openapi/openapi.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ info:
   license:
     name: Apache 2.0
     url: https://ignite.apache.org
-  version: 3.0.0-SNAPSHOT
+  version: 3.0.0-alpha
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/rest/openapi/openapi.yaml]: Oops, this thing again. Fixed it.

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/topology/LogicalTopologyImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/topology/LogicalTopologyImpl.java:32]: I was just curious, it's fine

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java [hunk 3]
```diff
@@ -245,11 +246,16 @@ private List<IgniteComponent> startPartialNode(
 
         var txManager = new TxManagerImpl(replicaService, lockManager, hybridClock);
 
+        var clusterStateStorage = new RocksDbClusterStateStorage(dir.resolve("cmg"));
+
+        LogicalTopologyImpl logicalTopologyService = new LogicalTopologyImpl(clusterStateStorage);
```
Comment (sashapolo) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java:251]: We can use `var` here as well. But this is not important

**Review by sashapolo — APPROVED** [review]

**Pull request merged**